File: python/6 kyu/401.py
# https://www.codewars.com/kata/6a5c17e763f7de3497eec2e2/train/python

# Passed

from collections.abc import Iterable
from typing_extensions import enum

class Hand(enum.IntEnum):
    NONE  = 0
    LEFT  = 1
    RIGHT = 2
    BOTH  = 3
(NONE,LEFT,RIGHT,BOTH) = Hand

def which_hand(word: Iterable[str]) -> Hand:
    left_hand_letters = set("qwertasdfgzxcvb")
    right_hand_letters = set("yuiophjklnm")
    left_right = [False, False]
    
    result = NONE
    for letter in word:
        if letter in left_hand_letters:
            left_right[0] = True
            result = LEFT
        elif letter in right_hand_letters:
            left_right[1] = True
            result = RIGHT
            
        if all(left_right):
            result = BOTH
            break
            
    return result

output = which_hand("cards")
print(output)

File: nodejs/7 kyu/401.js
// https://www.codewars.com/kata/5b0d67c1cb35dfa10b0022c7/train/javascript

// Passed

function squaresNeeded(grains) {
  let result = grains !== 0 ? Math.floor(Math.log2(grains)) + 1 : 0;
  return result;
}

const output = squaresNeeded(3);
console.log(output);